File: src/test/java/org/oewntk/xml/in/TestsXmlModelQueries.java
/*
 * Copyright (c) 2021-2021. Bernard Bou.
 */

package org.oewntk.xml.in;

import org.junit.BeforeClass;
import org.junit.Test;
import org.oewntk.model.LibTestModelQueries;
import org.xml.sax.SAXException;

import java.io.IOException;

import javax.xml.parsers.ParserConfigurationException;

public class TestsXmlModelQueries
{
	@BeforeClass
	public static void init() throws IOException, ParserConfigurationException, SAXException
	{
		TestsXmlCommon.init();
	}

	@Test
	public void testRowByType()
	{
		LibTestModelQueries.testWordByType(TestsXmlCommon.model, "row", TestsXmlCommon.ps);
	}

	@Test
	public void testRowByPos()
	{
		LibTestModelQueries.testWordByType(TestsXmlCommon.model, "row", TestsXmlCommon.ps);
	}

	@Test
	public void testRowByTypeAndPronunciation()
	{
		LibTestModelQueries.testWordByTypeAndPronunciation(TestsXmlCommon.model, "row", TestsXmlCommon.ps);
	}

	@Test
	public void testRowByPosAndPronunciation()
	{
		LibTestModelQueries.testWordByTypeAndPronunciation(TestsXmlCommon.model, "row", TestsXmlCommon.ps);
	}


	@Test
	public void testCriticalByType()
	{
		LibTestModelQueries.testWordByType(TestsXmlCommon.model, "critical", TestsXmlCommon.ps);
	}

	@Test
	public void testCriticalByPos()
	{
		LibTestModelQueries.testWordByType(TestsXmlCommon.model, "critical", TestsXmlCommon.ps);
	}

	@Test
	public void testBassByPos()
	{
		LibTestModelQueries.testWordByType(TestsXmlCommon.model, "bass", TestsXmlCommon.ps);
	}

	@Test
	public void testBaroqueByPos()
	{
		LibTestModelQueries.testWordByType(TestsXmlCommon.model, "baroque", TestsXmlCommon.ps);
	}

	@Test
	public void testBaroqueCSByPos()
	{
		LibTestModelQueries.testWordByType(TestsXmlCommon.model, "Baroque", TestsXmlCommon.ps);
	}

	@Test
	public void testGaloreByPos()
	{
		LibTestModelQueries.testWordByType(TestsXmlCommon.model, "galore", TestsXmlCommon.ps);
	}
}

File: src/test/java/org/oewntk/xml/in/TestsXmlModelQueries.kt
/*
 * Copyright (c) 2021-2021. Bernard Bou.
 */
package org.oewntk.xml.`in`

import org.junit.BeforeClass
import org.junit.Test
import org.oewntk.model.LibTestModelQueries.testWordByType
import org.oewntk.model.LibTestModelQueries.testWordByTypeAndPronunciation
import org.oewntk.xml.`in`.LibTestsXmlCommon.model
import org.oewntk.xml.`in`.LibTestsXmlCommon.ps

class TestsXmlModelQueries {
	@Test
	fun testRowByType() {
		testWordByType(model!!, "row", ps)
	}

	@Test
	fun testRowByPos() {
		testWordByType(model!!, "row", ps)
	}

	@Test
	fun testRowByTypeAndPronunciation() {
		testWordByTypeAndPronunciation(model!!, "row", ps)
	}

	@Test
	fun testRowByPosAndPronunciation() {
		testWordByTypeAndPronunciation(model!!, "row", ps)
	}


	@Test
	fun testCriticalByType() {
		testWordByType(model!!, "critical", ps)
	}

	@Test
	fun testCriticalByPos() {
		testWordByType(model!!, "critical", ps)
	}

	@Test
	fun testBassByPos() {
		testWordByType(model!!, "bass", ps)
	}

	@Test
	fun testBaroqueByPos() {
		testWordByType(model!!, "baroque", ps)
	}

	@Test
	fun testBaroqueCSByPos() {
		testWordByType(model!!, "Baroque", ps)
	}

	@Test
	fun testGaloreByPos() {
		testWordByType(model!!, "galore", ps)
	}

	companion object {
		@JvmStatic
		@BeforeClass
		fun init() {
			LibTestsXmlCommon.init()
			checkNotNull(model)
		}
	}
}
